qiskit/transpiler/target.py

```diff
@@ -0,0 +1,652 @@
+
+        Args:
+           gate (str): The gate instance to get qargs for
+        Returns:
+            set: The set of qargs the gate instance applies to
+        """
+        if None in self._gate_map:
+            return None
+        return self._gate_map[gate].keys()
+
+    def durations(self):
+        """Get an InstructionDurations object from the target
+
+        Returns:
+            InstructionDurations: The instruction duration represented in the
+                target
+        """
+        if self._instruction_durations is not None:
+            return self._instruction_durations
+        out_durations = []
+        for instruction, props_map in self._gate_map.items():
+            for qarg, properties in props_map.items():
+                if properties is not None and properties.duration is not None:
+                    out_durations.append((instruction, list(qarg), properties.duration, "s"))
+        self._instruction_durations = InstructionDurations(out_durations, dt=self.dt)
+        return self._instruction_durations
+
+    def timing_constraints(self):
+        """Get an :class:`~qiskit.transpiler.TimingConstraints` object from the target
+
+        Returns:
+            TimingConstraints: The timing constraints represented in the Target
+        """
+        return TimingConstraints(
+            self.granularity, self.min_length, self.pulse_alignment, self.aquire_alignment
+        )
+
+    def instruction_schedule_map(self):
+        """Return an :class:`~qiskit.pulse.InstructionScheduleMap` for the
+        instructions in the target with a pulse schedule defined.
+
+        Returns:
+            InstructionScheduleMap: The instruction schedule map for the
+            instructions in this target with a pulse schedule defined.
+        """
+        if self._instruction_schedule_map is not None:
+            return self._instruction_schedule_map
+        out_inst_schedule_map = InstructionScheduleMap()
+        for instruction, qargs in self._gate_map.items():
+            for qarg, properties in qargs.items():
+                if properties is not None and properties.calibration is not None:
+                    out_inst_schedule_map.add(instruction, qarg, properties.calibration)
+        self._instruction_schedule_map = out_inst_schedule_map
+        return out_inst_schedule_map
+
+    def operation_from_name(self, instruction):
+        """Get the operation class object for a given name
+
+        Args:
+            instruction (str): The instruction name to get the
+                :class:`~qiskit.circuit.Instruction` instance for
+        Returns:
+            Instruction: The Instruction instance corresponding to the name
+        """
+        return self._gate_name_map[instruction]
+
+    def operations_for_qargs(self, qarg):
+        """Get the operation class object for a specified qarg
+
+        Args:
+            qarg (tuple): A qarg tuple of the qubits to get the gates that apply
```

Comment:
`qargs` here as well?

Reply:
Here it's a single argument. Specifically like give me all the gates on `(0, 1)`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/599d2fda04ea02a54bba035b5aeaf82e1a123c2a